packages/x-charts/src/internals/plugins/utils/selectors.ts

```diff
@@ -1,11 +1,28 @@
 import { lruMemoize, createSelectorCreator, CreateSelectorFunction } from 'reselect';
+import { fastShallowCompare } from '@mui/x-internals/fastShallowCompare';
 import { ChartAnyPluginSignature, ChartState, ChartStateCacheKey } from '../models';
 
+const complainIfNoMemoize =
+  (name: string) =>
+  (a: any, b: any): boolean => {
+    const is = Object.is(a, b);
+    const shallow = fastShallowCompare(a, b);
+    if (is !== shallow) {
+      console.warn(
+        `Selector ${name} returned inconsistent results. Is: ${is} Shallow: ${shallow}`,
+        a,
+        b,
+      );
+    }
+    return is;
+  };
+
 const reselectCreateSelector = createSelectorCreator({
   memoize: lruMemoize,
   memoizeOptions: {
     maxSize: 1,
-    equalityCheck: Object.is,
+    equalityCheck: complainIfNoMemoize('prop check'),
+    resultEqualityCheck: complainIfNoMemoize('result check'),
```

Comment:
A possible solution is to just use `fastShallowCompare` everywhere. It should fixes all the issues we are seeing.

Another way is to use `process.env.NODE_ENV !== 'production' ? complainIfNoMemoize() : Object.is` and we will have to manually add the correct comparison to `useSelector`. Though this only fixes `result check`, not the `prop check`